src/librustdoc/html/static/js/source-script.js

```diff
@@ -33,7 +33,7 @@ function createDirEntry(elem, parent, fullPath, hasFoundFile) {
     folders.className = "folders";
     if (elem.dirs) {
         for (const dir of elem.dirs) {
-            if (createDirEntry(dir, folders, fullPath, hasFoundFile)) {
+            if (createDirEntry(dir, folders, fullPath, false)) {
```

Comment:
We definitely can! That should simplify the code a bit too.

Reply:
So no we can't: for `openParentDetails` to work, the DOM we created would need to be already inserted otherwise it does nothing... Just in case, here's the diff with this change:

```diff
diff --git a/src/librustdoc/html/static/js/main.js b/src/librustdoc/html/static/js/main.js
index 2aef978a072..8b79103397e 100644
--- a/src/librustdoc/html/static/js/main.js
+++ b/src/librustdoc/html/static/js/main.js
@@ -1,6 +1,6 @@
 // Local js definitions:
 /* global addClass, getSettingValue, hasClass, searchState */
-/* global onEach, onEachLazy, removeClass */
+/* global onEach, onEachLazy, removeClass, openParentDetails */
 
 "use strict";
 
@@ -353,15 +353,6 @@ function loadCss(cssFileName) {
         handleHashes(ev);
     }
 
-    function openParentDetails(elem) {
-        while (elem) {
-            if (elem.tagName === "DETAILS") {
-                elem.open = true;
-            }
-            elem = elem.parentNode;
-        }
-    }
-
     function expandSection(id) {
         openParentDetails(document.getElementById(id));
     }
diff --git a/src/librustdoc/html/static/js/source-script.js b/src/librustdoc/html/static/js/source-script.js
index a6a0b09ef31..156b1ac1d7d 100644
--- a/src/librustdoc/html/static/js/source-script.js
+++ b/src/librustdoc/html/static/js/source-script.js
@@ -3,7 +3,7 @@
 
 // Local js definitions:
 /* global addClass, getCurrentValue, onEachLazy, removeClass, browserSupportsHistoryApi */
-/* global updateLocalStorage */
+/* global updateLocalStorage, openParentDetails */
 
 "use strict";
 
@@ -18,7 +18,7 @@ function closeSidebarIfMobile() {
     }
 }
 
-function createDirEntry(elem, parent, fullPath, hasFoundFile) {
+function createDirEntry(elem, parent, fullPath) {
     const dirEntry = document.createElement("details");
     const summary = document.createElement("summary");
 
@@ -33,10 +33,7 @@ function createDirEntry(elem, parent, fullPath, hasFoundFile) {
     folders.className = "folders";
     if (elem.dirs) {
         for (const dir of elem.dirs) {
-            if (createDirEntry(dir, folders, fullPath, false)) {
-                dirEntry.open = true;
-                hasFoundFile = true;
-            }
+            createDirEntry(dir, folders, fullPath);
         }
     }
     dirEntry.appendChild(folders);
@@ -50,17 +47,15 @@ function createDirEntry(elem, parent, fullPath, hasFoundFile) {
             file.href = rootPath + "src/" + fullPath + file_text + ".html";
             file.addEventListener("click", closeSidebarIfMobile);
             const w = window.location.href.split("#")[0];
-            if (!hasFoundFile && w === file.href) {
+            if (w === file.href) {
                 file.className = "selected";
-                dirEntry.open = true;
-                hasFoundFile = true;
+                openParentDetails(file);
             }
             files.appendChild(file);
         }
     }
     dirEntry.appendChild(files);
     parent.appendChild(dirEntry);
-    return hasFoundFile;
 }
 
 function toggleSidebar() {
@@ -118,16 +113,13 @@ function createSourceSidebar() {
     const sidebar = document.createElement("div");
     sidebar.id = "source-sidebar";
 
-    let hasFoundFile = false;
-
     const title = document.createElement("div");
     title.className = "title";
     title.innerText = "Files";
     sidebar.appendChild(title);
     Object.keys(sourcesIndex).forEach(key => {
         sourcesIndex[key].name = key;
-        hasFoundFile = createDirEntry(sourcesIndex[key], sidebar, "",
-            hasFoundFile);
+        createDirEntry(sourcesIndex[key], sidebar, "");
     });
 
     container.appendChild(sidebar);
diff --git a/src/librustdoc/html/static/js/storage.js b/src/librustdoc/html/static/js/storage.js
index 0c5389d45e5..0ce411a67a8 100644
--- a/src/librustdoc/html/static/js/storage.js
+++ b/src/librustdoc/html/static/js/storage.js
@@ -67,6 +67,15 @@ function removeClass(elem, className) {
     elem.classList.remove(className);
 }
 
+function openParentDetails(elem) {
+    while (elem) {
+        if (elem.tagName === "DETAILS") {
+            elem.open = true;
+        }
+        elem = elem.parentNode;
+    }
+}
+
 /**
  * Run a callback for every element of an Array.
  * @param {Array<?>}    arr        - The array to iterate over
```